authors/apps/articles/views.py

```diff
@@ -43,3 +50,46 @@ def destroy(self, request, *args, **kwargs):
         self.perform_destroy(instance)
         return Response({"message": "category deleted"},
                         status=status.HTTP_204_NO_CONTENT)
+
+
+class CreateArticle(ListCreateAPIView):
+    permission_classes = (IsAuthenticated,)
+    renderer_classes = (ArticleJSONRenderer,)
+    serializer_class = ArticleSerializer
+    queryset = Article.objects.all()
+
+    def create(self, request):
+
+        article = request.data.get('article', {})
+        current_user = User.objects.get(email=request.user)
+        serializer = self.get_serializer(data=article)
+        serializer.is_valid(raise_exception=True)
+        serializer.save(author_id=current_user.id,
```

Comment:
@Stanley-Okwii You can do this as commented above and you don't need to pass the category_id parameter since **category** is part of the request body.

Reply:
@PatrickCmd , category_id received from the request body is a category slug which helps relate a category to an article. I have implemented as you mentioned ealrier.